### ISSUE TITLE
[Compact Profile 2] - core.scheduler bundle depends on Full JRE

### ISSUE THREAD

**Issue by firasalmanna** [eclipse/smarthome#5806]
I am working on running the ESH framework on a compact2 device. I found that the `org.eclipse.smarthome.core.scheduler` bundle requires Full JRE. It has a dependency on `org.quartz.impl.StdSchedulerFactory` class [1] which depends on `java.beans` packages [2], and the Java Beans API is available only in the Full JRE.

I disable the scheduler bundle and the framework is running now, but I am not sure about the effects of disabling it as my experience with the framework is still new.

[1]
https://github.com/eclipse/smarthome/blob/7176f8961a4caba501fe38bab52a44c8705011c3/bundles/core/org.eclipse.smarthome.core.scheduler/src/main/java/org/eclipse/smarthome/core/scheduler/internal/SchedulerActivator.java#L18
[2]
https://github.com/quartz-scheduler/quartz/blob/9f9e400733f51f7cb658e3319fc2c140ab8af938/quartz-core/src/main/java/org/quartz/impl/StdSchedulerFactory.java#L54

**Comment by kaikreuzer** [comment]
You are right, Quartz is one of the remainders to get rid off for proper CP2 support - note that https://github.com/eclipse/smarthome/pull/4295 will take care of this.
